powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java

```diff
@@ -96,13 +98,17 @@ public BasePushMessageSendResult sendPushMessage(final String appId, final Mode
 
                     final Platform platform = device.getPlatform();
                     if (platform == Platform.IOS || platform == Platform.APNS) {
-                        if (pushClient.getApnsClient() == null) {
-                            logger.error("Push message cannot be sent to APNS because APNS is not configured in push server.");
+                        final PushMessageSendResult.PlatformResult platformResult = sendResult.getApns();
+                        final PushSendingCallback callback = createPushSendingCallback(mode, device, platformResult, pushMessageObject, phaser);
+                        final String environment = resolveApnsEnvironment(device.getEnvironment(), appCredentials.getApnsEnvironment());
+                        if (!configuration.isApnsUseDevelopment() && ApnsEnvironment.DEVELOPMENT.getKey().equals(device.getEnvironment())) {
```

Comment:
This will not take into the account server appCredentials setting.

we can simply as:
```
// if the device environment is used, and the device environment is not the same as resolved environment we have to skip.
if (device.getEnvironment() !=null && !device.getEnvironment().equals(environment))
```

Alternatively you can modify resolveEnvironment method and ensure null is returned in the case of incorrect state.

Example in resolveEnvironment:

```
        if (environmentDevice != null) {
            if (environmentDevice.equals(ApnsEnvironment.PRODUCTION.getKey()) || environment.equals(ApnsEnvironment.DEVELOPMENT.getKey())) {
                environment = environmentDevice;
            } else {
               // environment cannot be resolved as we have conflicting server settings and device settings
               environment = null;
           }
        }
```

Reply:
Understood. I prefer the variant with returning `null` value and handling this state outside the method.

Reply:
I added a test for this case. Both global configuration and application credential configuration setting is tested.